src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -119,11 +152,10 @@ public function remove($name)
      */
     public function addCollection(RouteCollection $collection)
     {
-        // we need to remove all routes with the same names first because just replacing them
-        // would not place the new route at the end of the merged array
-        foreach ($collection->all() as $name => $route) {
-            unset($this->routes[$name]);
-            $this->routes[$name] = $route;
+        foreach ($collection->prioritizedRoutes as $priority => $routes) {
```

Comment:
this breaks classes extending `RouteCollection` - we might need a new method to get routes by priorities